src/OpenTelemetry.Api/Trace/ActivityExtensions.cs

```diff
@@ -73,5 +75,29 @@ public static Status GetStatus(this Activity activity)
 
             return status;
         }
+
+        /// <summary>
+        /// Sets the kind of activity execution.
+        /// </summary>
+        /// <param name="activity">Activity instance.</param>
+        /// <param name="kind">Activity execution kind.</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static void SetKind(this Activity activity, ActivityKind kind)
+        {
+            Debug.Assert(activity != null, "Activity should not be null");
+            setKindProperty(activity, kind);
+        }
+
+#pragma warning disable SA1201 // Elements should appear in the correct order
+        private static Action<Activity, ActivityKind> setKindProperty = CreateActivityKindSetter();
```

Comment:
@eddynaka Would you mind adding `readonly` to this?

Reply:
done!